File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.cpp
class Solution {
public:
    int maxIceCream(vector<int>& costs, int coins) {
        priority_queue<int, vector<int>, greater<int>> pq;
        for (auto x : costs) pq.push(x);

        int ans = 0;
        while (!pq.empty()) {
            int curr = pq.top();
            pq.pop();
            if(coins < curr) break;
            ans++; coins -= curr;

        }

        return ans;
    }
};

File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.ts
function maxIceCream(costs: number[], coins: number): number {
    costs.sort((a, b) => a - b);
    let i = 0;
    for (i = 0; i < costs.length; i++) {
        if (coins - costs[i] < 0) break;
        coins -= costs[i];
    }
    return i;
};